File: android/app/src/main/java/com/expensetracker/ui/components/Cards.kt
package com.expensetracker.ui.components

import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.*
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.shadow
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.tooling.preview.Preview
import androidx.compose.ui.unit.dp
import com.expensetracker.ui.theme.*

@Composable
fun StatTile(
    title: String,
    amount: String,
    modifier: Modifier = Modifier,
    color: Color = PrimaryBlue
) {
    PastelCard(
        modifier = modifier
    ) {
        Column(
            horizontalAlignment = Alignment.CenterHorizontally,
            verticalArrangement = Arrangement.Center,
            modifier = Modifier.fillMaxWidth()
        ) {
            Text(
                text = title,
                style = MaterialTheme.typography.titleSmall,
                color = MaterialTheme.colorScheme.onSurface.copy(alpha = 0.7f)
            )
            
            Spacer(modifier = Modifier.height(4.dp))
            
            Text(
                text = amount,
                style = MaterialTheme.typography.titleMedium,
                fontWeight = FontWeight.SemiBold,
                color = color
            )
        }
    }
}

@Composable
fun PastelCard(
    modifier: Modifier = Modifier,
    content: @Composable ColumnScope.() -> Unit
) {
    Card(
        modifier = modifier
            .fillMaxWidth()
            .shadow(
                elevation = 2.dp,
                shape = RoundedCornerShape(12.dp),
                ambientColor = Color.Black.copy(alpha = 0.05f),
                spotColor = Color.Black.copy(alpha = 0.05f)
            )
            .border(
                width = 1.dp,
                color = CardBorder,
                shape = RoundedCornerShape(12.dp)
            ),
        colors = CardDefaults.cardColors(
            containerColor = CardBackground
        ),
        shape = RoundedCornerShape(12.dp),
        elevation = CardDefaults.cardElevation(defaultElevation = 0.dp)
    ) {
        Column(
            modifier = Modifier.padding(16.dp),
            content = content
        )
    }
}

@Preview(showBackground = true)
@Composable
fun CardsPreview() {
    ExpenseTrackerTheme {
        Column(
            modifier = Modifier.padding(16.dp),
            verticalArrangement = Arrangement.spacedBy(16.dp)
        ) {
            Row(
                horizontalArrangement = Arrangement.spacedBy(12.dp)
            ) {
                StatTile(
                    title = "Total Income",
                    amount = "$12,500",
                    modifier = Modifier.weight(1f),
                    color = IncomeGreen
                )
                StatTile(
                    title = "Total Expenses",
                    amount = "$8,200",
                    modifier = Modifier.weight(1f),
                    color = ExpenseRed
                )
            }
            
            StatTile(
                title = "Balance",
                amount = "$4,300",
                color = AccentSuccess
            )
            
            PastelCard {
                Column(
                    verticalArrangement = Arrangement.spacedBy(8.dp)
                ) {
                    Text(
                        text = "Sample Card",
                        style = MaterialTheme.typography.headlineMedium
                    )
                    Text(
                        text = "This is a sample card with pastel styling",
                        style = MaterialTheme.typography.bodyLarge,
                        color = MaterialTheme.colorScheme.onSurface.copy(alpha = 0.7f)
                    )
                }
            }
        }
    }
}

File: ios/ExpenseTracker/Components/Cards.swift
import SwiftUI

// MARK: - Stat Tile

struct StatTileView: View {
    let title: String
    let amount: String
    let color: Color = Theme.Colors.primary
    
    var body: some View {
        VStack(spacing: Theme.Spacing.xs) {
            Text(title)
                .font(Theme.Typography.statTitle)
                .foregroundColor(Theme.Colors.onSurface.opacity(0.7))
            
            Text(amount)
                .font(Theme.Typography.statValue)
                .fontWeight(.semibold)
                .foregroundColor(color)
        }
        .padding(Theme.Spacing.lg)
        .frame(maxWidth: .infinity)
        .background(
            RoundedRectangle(cornerRadius: Theme.Shapes.cardCornerRadius)
                .fill(Theme.Colors.card)
                .shadow(color: .black.opacity(0.05), radius: Theme.Elevation.low, x: 0, y: 1)
                .overlay(
                    RoundedRectangle(cornerRadius: Theme.Shapes.cardCornerRadius)
                        .stroke(Theme.Colors.cardBorder, lineWidth: 1)
                )
        )
    }
}

// MARK: - Card Container

struct PastelCard<Content: View>: View {
    let content: Content
    
    init(@ViewBuilder content: () -> Content) {
        self.content = content()
    }
    
    var body: some View {
        content
            .padding(Theme.Spacing.lg)
            .background(
                RoundedRectangle(cornerRadius: Theme.Shapes.cardCornerRadius)
                    .fill(Theme.Colors.card)
                    .shadow(color: .black.opacity(0.05), radius: Theme.Elevation.low, x: 0, y: 1)
                    .overlay(
                        RoundedRectangle(cornerRadius: Theme.Shapes.cardCornerRadius)
                            .stroke(Theme.Colors.cardBorder, lineWidth: 1)
                    )
            )
    }
}

#Preview {
    VStack(spacing: 16) {
        HStack(spacing: 12) {
            StatTileView(title: "Total Income", amount: "$12,500")
            StatTileView(title: "Total Expenses", amount: "$8,200", color: Theme.Colors.expense)
        }
        
        StatTileView(title: "Balance", amount: "$4,300", color: Theme.Colors.success)
        
        PastelCard {
            VStack(alignment: .leading, spacing: 8) {
                Text("Sample Card")
                    .font(Theme.Typography.titleM)
                Text("This is a sample card with pastel styling")
                    .font(Theme.Typography.body)
                    .foregroundColor(Theme.Colors.onSurface.opacity(0.7))
            }
        }
    }
    .padding()
    .background(Theme.Colors.surface)
}